src/kvstore.c

```diff
@@ -173,11 +173,18 @@ static dict *createDictIfNeeded(kvstore *kvs, int didx) {
     return kvs->dicts[didx];
 }
 
+/* Called when the dict will delete entries, the function will check
+ * KVSTORE_FREE_EMPTY_DICTS to determine whether the empty dict needs
+ * to be freed.
+ *
+ * Note that for rehashing dicts, that is, in the case of safe iterators,
+ * we won't delete the dict. The dict will check whether it needs to
```

Comment:
```suggestion
 * we won't delete the dict. We will check whether it needs to
```

Reply:
i still feel that it might be a good idea to drop a short comment when we call this method to explain why.
like you did for scan.